CMakeLists.txt

```diff
@@ -94,6 +94,36 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 message(STATUS "Supported C compiler features = ${CMAKE_C_COMPILE_FEATURES}")
 message(STATUS "Supported C++ compiler features = ${CMAKE_CXX_COMPILE_FEATURES}")
 
+# Enable some warnings if we know the compiler.
+if(MSVC)
+  add_compile_options(/W4 /analyze)
```

Comment:
we might need to check msvc version for `/analyze`

Reply:
Do we though? It works on 2019. Do we want to support versions lower than that?

Reply:
lets leave it as is until someone complains